src/DFG.cpp

```diff
@@ -1202,6 +1255,12 @@ void DFG::replaceDFGEdge(DFGNode* t_old_src, DFGNode* t_old_dst,
   m_DFGEdges.push_back(newEdge);
 }
 
+void DFG::addDFGEdge(DFGNode* t_src, DFGNode* t_dst){ // combineForUnroll
+  int newEdgeID = (m_DFGEdges.back())->getID();
+  DFGEdge* newEdge = new DFGEdge(newEdgeID, t_src, t_dst);
```

Comment:
Shouldn't the `int newEdgeID = (m_DFGEdges.back())->getID() + 1;`?
Can you make this as a private function in DFG class? Say, `int getNewEdgeID() { return (m_DFGEdges.back())->getID() + 1;}`?